README
README
======

Download
----------------------

Das Modul wird Open Source unter der Open Software License (OSL 3.0) angeboten und kann unter folgender URL kostenlos heruntergeladen werden:

http://www.touchdesign.de/loesungen/prestashop/sofortueberweisung.htm

Autor & Copyright
------------------

Copyright by: touchdesign
Chief developer: Christin Gruber

Licenses
--------

Released under Open Software License (OSL 3.0).

Installation
-------------------------------------

A) Upload

Wechseln Sie in Ihrem Shopsystem über den "Adminbereich" in den Tab "Module" und klicken Sie auf den
Link "Ein neues Modul hinzufügen". Klicken Sie auf den Button "Durchsuchen" und wählen Sie nun das 
zuvor heruntergeladene Modularchiv "sofortbanking-vX.tar" von Ihrer lokalen Festplatte.
Nach dem Sie das Modularchiv angewählt haben bestätigen Sie bitte die Installation des Modules durch einen 
Klick auf den Button "Dieses Modul hochladen". Nach erfolgreicher Installation des Modules erscheint die 
Meldung "Modul wurde erfolgreich installiert".

B) Modul Konfiguration

Nach erfolgreicher Installation des Modules wechseln Sie bitte in die Modul-Konfiguration durch Anwahl 
des soeben installierten Modules "sofortueberweisung.de" in der Modulliste unter der Kategorie "Payment" durch einen Klick 
auf den Link "Einstellungen". Die Paramater "User ID", "Projekt ID", "Projekt Passwort" und "Benachrichtigungs-Passwort" sind 
durch die von der Payment Network AG zugeteilten Kunden- sowie Projektdaten zu befüllen.

C) Schnittstellen und Projektkonfiguration

Bitte führen Sie folgende Schritte in Ihrem Kundenlogin auf sofortueberweisung.de durch um die 
Konfiguration der Schnittstelle zu vervollständigen:

  1) Sofern Sie noch kein Projekt für Ihren Onlineshop eingerichtet haben, legen Sie dieses bitte zuvor an.
  
  2) Bitte wählen Sie im Kundenlogin auf sofortueberweisung.de bei der Projektkonfiguration unter "Allgemeine Einstellungen" das Shopsystem "PrestaShop" aus der Dropdownliste. 

  3) Hinterlegen Sie bitte die im Modul generierte "Erfolgs-URL (Weiterleitung zur Bestätigung)" für Ihren Shop in der Projektkonfiguration unter "Schnittstelle > Erfolgslink".

  4) Hinterlegen Sie bitte die im Modul generierte "Abbruchs-URL (Rückkehr zum Shop)" für Ihren Shop in der Projektkonfiguration unter "Schnittstelle > Abbruch-Link".

  5) Richten Sie die HTTP Benachrichtigung (!nicht HTTP-alt) unter dem Tab "Erweiterte Einstellungen > Benachrichtgungen" ein und übernehmen Sie die "Bestätigungs-URL (Http-Benachrichtigung)" welche unter den Moduleinstellungen generiert wird. In der Auswahl für die Request "Methode" wählen Sie bitte "POST".

  6) Vervollständigen Sie die weiteren Angaben wie Adresse, Shopname und Bankdaten um die Konfiguration abzuschliessen.

Nach erfolgreichem Anlegen Ihres Projektes finden Sie weitere detaillierte Informationen zur Konfiguration 
der Schnittstelle unter dem Karteireiter "Hilfe zum Shopsystem" in Ihrem Kundenlogin auf sofortueberweisung.de.

Ressourcen
-------------------------------------

[1]: http://www.touchdesign.de/loesungen/prestashop/sofortueberweisung.htm

CHANGELOG
// Version 258

Add backward compatibility for versions lower than 1.7
Add compatibility for PS 1.7
Remove old payment_redirect.tpl
Fix success validation
Update missing translations
Fix change order state
Fix validation bug, thx for the feedback to Marek Vavrecan
Fix trim white spaces for input hash

// Version 2571

Fix trim white spaces for input hash

// Version 257

Update translations
Fetch front-end images directly from sofort server
Remove unused local images

// Version 256

Increment...

// Version 255

Fix right/left column on payment

// Version 253

Update hungarian translation
Update module version

// Version 24

Increment...

// Version 23

Add order state dropdowns for accepted and failed payments
Add optional checkbox to ignore status update events
Set new default status for payed orders
Small fixes for PS coding standards
Add order validation by confirmation url
Update notification validation.php for failed payments
Fix token redirect and authentication check in payment.php for PS compat 1.4
Add compats for PS 1.6
Add new clear module configuration layout
Update translations
Add new translations (Slovak, Czech, Hungarian)
Fix invalid token in one page checkout
Fix urls for multistores

// Version 22

Increment...

// Version 21

Increment...

// Version 20

Remove old Payment tab
Replace globals by Tools::getValue()
Remove unused code from touchdesign lib
Move template files to views
Rewrite admin display form for views
Cast some user values for "better performances"...
Remove some unused code for old backwards
Add token for payment.php
Code cleanup for PrestaShop coding standards
Add backward_compatibility
Replace ucfrist with Tools::ucfirst
Escape template vars
Replace strtolower with Tools::strtolower
Add some compatibility fixes for PS 1.4
Add extra string for payment button

// Version 19

Add currency paramater for CHF,GBP and Euro currencies
Add module lang fallback
Add new translation for gb
Fix translation strings

// Version 18

Small fix to grep backend language with or without context
Add index.php protection for sub-directories
Add some compats in dev_mode for PS 1.5
Add controllers and views for PS 1.5
Replace spaces by tabs
Code cleanup
Fix cancellation link for PS 1.5

// Version 17

Add backward compatibility for PS 1.4

// Version 16

Add compatibility for PS 1.5.4
Fix payment tpl {l ...} tag
Add multi language support for images
Add new translations for es, fr, it, nl and pl
Fix module name translation

// Version 15

Add compatibility for PS 1.5.3
Update translations
Switch icon

// Version 14

Logo update

// Version 13

Add compatibility for PS 1.5

// Version 12

Add optional force redirect (usefull for non EU and modified stores)

// Version 11

...

// Version 10

Update README
Update Translations
Update logos
Rename sofortueberweisung -> sofortbanking
Add secure_key

// Version 09

Fix guest checkout
Add touchdesign library
Fix redirects

// Version 08

Small changes for prestashop version 1.4 compatibility
Add confirm before payment page
Update README
Update Translations

// Version 07

Update cancel url
Add readme/install doc
Update translation
Add payment block
Add success message
Add required note
Add confirmation page
Update order states
Code cleanup
Add changelog
Add customer protection by sofort bank

// ...

payment.php
<?php
/**
 * $Id$
 *
 * sofortbanking Module
 *
 * Copyright (c) 2009 touchdesign
 *
 * @category Payment
 * @version 2.0
 * @copyright 19.08.2009, touchdesign
 * @author Christin Gruber, <www.touchdesign.de>
 * @link http://www.touchdesign.de/loesungen/prestashop/sofortueberweisung.htm
 * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
 *
 * Description:
 *
 * Payment module sofortbanking
 *
 * --
 *
 * NOTICE OF LICENSE
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.
 * It is also available through the world-wide-web at this URL:
 * http://opensource.org/licenses/osl-3.0.php
 * If you did not receive a copy of the license and are unable to
 * obtain it through the world-wide-web, please send an email
 * to license@touchdesign.de so we can send you a copy immediately.
 *
 */

/**
 * Notice: This file will be removed in new module version for PS 1.6
 */

require_once dirname(__FILE__).'/../../config/config.inc.php';
require_once dirname(__FILE__).'/sofortbanking.php';

/* If PS 1.5 or higher redirect to module controller */
if (version_compare(_PS_VERSION_, '1.5', '>='))
	Tools::redirect(Context::getContext()->link->getModuleLink('sofortbanking', 'payment').'?token='.Tools::getValue('token'));
else
{

	$controller = new FrontController();
	$controller->init();
	$controller->auth = true;

	/* Check if token is valid */
	if (Configuration::get('PS_TOKEN_ENABLE') && !(strcasecmp(Tools::getToken(false), Tools::getValue('token')) == 0))
		Tools::redirect('order.php', 'order=back.php');

	$controller->setMedia();
	$controller->displayHeader();

	$sofortbanking = new sofortbanking();

	/* Build and display payment page */
	echo $sofortbanking->backwardPaymentController();

	$controller->displayFooter();

}

?>